standalone_self_upgrade.cf.in

```diff
@@ -70,17 +70,6 @@ bundle agent cfengine_software
   vars:
 
       # Default desired CFEngine software
-
-      "pkg_name" string => "cfengine-nova";
-      "pkg_version" string => "@VERSION@";
-      "pkg_release" string => "1";
-      "pkg_arch" string => "x86_64";
-      "package_dir" string => "$(sys.flavour)_$(sys.arch)";
-
-!(cfengine_3_7_1|cfengine_3.7.2|cfengine_3.7.3)::
-# After 3.7.4 a fix to ifelse and isvarible allows for actuating the function
-# even thought the promise references an unresolved variable.
```

Comment:
Isn't this broken now? :smile:

Reply:
Broken?

It was 3.7.4 when ifelse() recieved a fix to implement special behaviour to avoid skipping if exactly 3 parameters are used in conjunction with unresolved variables.

The case being like this. Here pkg_name is set to def.cfengine_software_pkg_name if its defined, else to cfengine-nova. Without the fix recieved in 3.7.4 this promise was skipped.

```
       "pkg_name"
         string => ifelse( isvariable( "def.cfengine_software_pkg_name" ),
                                  $(def.cfengine_software_pkg_name),
                                  "cfengine-nova");
```

I think what is currently "broken" is that it doesn't work when you have more than 3 parameters.

So, for example, I can't check multiple variables in some preferred order. This promise would be skipped if either cfengine_internal_software_name OR def.cfengine_software_pkg_name are not defined.

```
       "pkg_name"
         string => ifelse( isvariable( "def.cfengine_software_pkg_name" ),
                                  $(def.cfengine_software_pkg_name),
                                  isvariable( "cfengine_internal.software_name"),
                                  $(cfengine_internal.software_name),
                                  "cfengine-nova");
```

Or are you referring to something else?

Reply:
> Or are you referring to something else?

Probably what you described, I wasn't sure. Just recalled that you mentioned this being broken in some specific case.

Reply:
Ah, but i did go and hunt down a ticket that notes a regression related to ifelse()

https://tracker.mender.io/browse/ENT-4653